YubiKit/YubiKit/FIDO/WebAuthn/AuthenticatorData.swift

```diff
@@ -0,0 +1,215 @@
+
+        /// Backup State (BS): Credential is currently backed up.
+        static let backupState = Flags(rawValue: 1 << 4)
+
+        /// Attested Credential Data (AT): Includes attested credential data.
+        static let attestedCredentialData = Flags(rawValue: 1 << 6)
+
+        /// Extension Data (ED): Includes extension data.
+        static let extensionData = Flags(rawValue: 1 << 7)
+    }
+}
+
+// MARK: - Binary Parsing
+
+extension AuthenticatorData {
+    /// Parse authenticator data from raw bytes.
+    ///
+    /// - Parameter data: The raw authenticator data bytes.
+    /// - Returns: Parsed authenticator data, or nil if parsing fails.
+    init?(data: Data) {
+        // Minimum size: rpIdHash (32) + flags (1) + signCount (4) = 37 bytes
+        guard data.count >= 37 else {
+            return nil
+        }
+
+        var offset = 0
+
+        // MARK: Parse RP ID Hash
+        self.rpIdHash = data.subdata(in: offset..<(offset + 32))
+        offset += 32
+
+        // MARK: Parse Flags
+        let flagsByte = data[offset]
+        self.flags = Flags(rawValue: flagsByte)
+        offset += 1
+
+        // MARK: Parse Signature Counter
+        let signCountData = data.subdata(in: offset..<(offset + 4))
+        self.signCount = signCountData.withUnsafeBytes { $0.loadUnaligned(as: UInt32.self).bigEndian }
+        offset += 4
+
+        // MARK: Parse Attested Credential Data (optional)
+        if flags.contains(.attestedCredentialData) {
+            guard let (attestedData, newOffset) = AttestedCredentialData.parse(from: data, startingAt: offset) else {
+                return nil
+            }
+            self.attestedCredentialData = attestedData
+            offset = newOffset
+        } else {
+            self.attestedCredentialData = nil
+        }
+
+        // MARK: Parse Extensions (optional)
+        if flags.contains(.extensionData) {
+            // Extensions are CBOR-encoded
+            let extensionsData = data.subdata(in: offset..<data.count)
+            guard let extensionsValue: CBOR.Value = try? extensionsData.decode(),
+                let extensions = ExtensionOutputs(cbor: extensionsValue)
+            else {
+                return nil
+            }
+            self.extensions = extensions
+        } else {
+            self.extensions = nil
+        }
+    }
+}
+
+// MARK: - Attested Credential Data
+
+/// Attested credential data included in authenticator data during credential creation.
+///
+/// Contains the AAGUID, credential ID, and credential public key.
+///
+/// - SeeAlso: [WebAuthn Attested Credential Data](https://www.w3.org/TR/webauthn/#sctn-attested-credential-data)
+struct AttestedCredentialData: Sendable {
+    /// Authenticator Attestation GUID (16 bytes).
+    let aaguid: Data
+
+    /// The credential ID (variable length).
+    let credentialId: Data
+
+    /// The credential public key in COSE format (CBOR-encoded).
+    let credentialPublicKey: Data
```

Comment:
Would it make sense to introduce a `COSEKey` type here that is essentially a CBOR map, but could expose type-safe getters for kty, kid, alg?
See https://datatracker.ietf.org/doc/html/rfc8152#section-7